Loader/Library/CopyService.cs

```diff
@@ -52,6 +52,36 @@ private async Task IterateDirectory(Logger logger, DownloadDirectory directory,
     private static string GetCuraLocation(string? version) =>
         Path.Join(GetAppDataDirectory(), "cura", version);
 
+    private static bool IsUnixLike()
+    {
+        // Should return true if the system behaves like a
+        // "traditional" Unix system.
+        return (OperatingSystem.IsLinux() || OperatingSystem.IsFreeBSD());
+    }
+
+    private static string GetUnixAppDataDirectory()
+    {
+        string? xdgDataHomeMaybe = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
```

Comment:
Is there any reason we should prefer the hard coded environment variables over the library's built in SpecialFolder enum like the Windows / macOS implementations below? For reference there's a chart of some of the options here:
https://developers.redhat.com/blog/2018/11/07/dotnet-special-folder-api-linux

Reply:
The main reason was that I didn't know about it. It wasn't clear to me at the time what the enums mapped to on Linux.

Reply:
I've updated my code to fit with your suggested changes. For some reason, there's also a commit "[Merge branch 'phnxdrgn536:main' into main](https://github.com/phnxdrgn536/FLSun-v400/pull/27/commits/f3356c9e582abc4732ce805afefab2e834e930fb), but I don't really know what that means. I hope it's harmless.

Reply:
Looks like changes upstream making it into your branch. Nothing to worry about.